client/blocks/stats-navigation/index.js

```diff
@@ -221,7 +224,19 @@ class StatsNavigation extends Component {
 					AVAILABLE_PAGE_MODULES[ this.props.selectedItem ] &&
 					! hideModuleSettings && (
 						<PageModuleToggler
-							availableModules={ AVAILABLE_PAGE_MODULES[ this.props.selectedItem ] }
+							availableModules={ AVAILABLE_PAGE_MODULES[ this.props.selectedItem ].map(
+								( toggleItem ) => {
+									// disable the "videos" toggle on sites that do not have VideoPress enabled
+									// the toggle will be disabled (grayed out and non interactive)
+									const shouldDisableVideoToggle = ! isVideoPress && toggleItem.key === 'videos';
+
+									return {
+										...toggleItem,
+										disabled: shouldDisableVideoToggle,
+										defaultValue: shouldDisableVideoToggle === false,
+									};
+								}
+							) }
```

Comment:
I’d recommend moving this logic out of the render() method, as it creates a new array on every render. This change will improve both performance (by preventing unnecessary re-renders) and maintainability (by keeping the render method focused on rendering).

I think we can leverage `getDerivedStateFromProps`, which is invoked during the initial render and whenever the component’s props change. This allows us to update the `disabled` and `defaultValue` properties of the toggle dynamically whenever the `isVideoPress` prop changes.

Here’s how it could look (I renamed `shouldDisableVideoToggle` to `shouldDisableToggle` to make it generic):

```
static getDerivedStateFromProps( nextProps, prevState ) {
		const availableModuleToggles = AVAILABLE_PAGE_MODULES[ nextProps.selectedItem ].map(
			( toggleItem ) => {
				const shouldDisableToggle = ! nextProps.isVideoPress && toggleItem.key === 'videos';

				return {
					...toggleItem,
					disabled: shouldDisableToggle,
					defaultValue: ! shouldDisableToggle,
				};
			}
		);

		if (
			prevState.pageModules !== nextProps.pageModuleToggles ||
			JSON.stringify( prevState.availableModuleToggles ) !== JSON.stringify( availableModuleToggles )
		) {
			return { availableModuleToggles, pageModules: nextProps.pageModuleToggles };
		}

    return null;
}
```

We also need to initialize availableModuleToggles in the component state with a default value (e.g., an empty array). This ensures no issues occur before getDerivedStateFromProps is invoked:

```
state = {
    availableModuleToggles: [], // Initialized as an empty array
    (...)
};
````

Reply:
See #96894